dbms/src/TiDB/Schema/SchemaBuilder.cpp

```diff
@@ -1320,23 +1341,21 @@ void SchemaBuilder<Getter, NameMapper>::syncAllSchema()
         = ThreadPool(default_num_threads, default_num_threads / 2, default_num_threads * 2);
     auto sync_all_schema_wait_group = sync_all_schema_thread_pool.waitGroup();
 
-    std::mutex created_db_set_mutex;
-    std::unordered_set<String> created_db_set;
+    SyncedDatabaseSet latest_db_nameset;
```

Comment:
```suggestion
    SyncedDatabaseSet latest_db_nameset;
    databases.clear();
```

Maybe just clear cache here?

Reply:
I think clear cache and let those existing database re-created by the loop is hard to understand.

Just let the `nameset` record all the database name of database info fetch from tikv, and dropped the local database that is not exists in the nameset is more easy to understand.